kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/lookup/EntryBoHelperServiceImpl.java

```diff
@@ -0,0 +1,323 @@
+                    String inquiryURL = generateInquiryURL(row, field, keys);
+
+                    HtmlData columnAnchor = new HtmlData.AnchorHtmlData(inquiryURL, title);
+                    column.setColumnAnchor(columnAnchor);
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private static Class getClassForField(String field) {
+        switch (field) {
+            case KFSPropertyConstants.GEC_DOCUMENT_NUMBER:
+            case KFSPropertyConstants.DOCUMENT_NUMBER:
+                return null;
+            case KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR:
+                return org.kuali.kfs.sys.businessobject.SystemOptions.class;
+            case KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE:
+                return org.kuali.kfs.coa.businessobject.Chart.class;
+            case KFSPropertyConstants.FINANCIAL_SYSTEM_ORIGINATION_CODE:
+            case KFSPropertyConstants.REFERENCE_FINANCIAL_SYSTEM_ORIGINATION_CODE:
+                return org.kuali.kfs.sys.businessobject.OriginationCode.class;
+            case KFSPropertyConstants.ACCOUNT_NUMBER:
+                return org.kuali.kfs.coa.businessobject.Account.class;
+            case KFSPropertyConstants.SUB_ACCOUNT_NUMBER:
+                return org.kuali.kfs.coa.businessobject.SubAccount.class;
+            case KFSPropertyConstants.FINANCIAL_OBJECT_CODE:
+                return org.kuali.kfs.coa.businessobject.ObjectCode.class;
+            case KFSPropertyConstants.FINANCIAL_SUB_OBJECT_CODE:
+                return org.kuali.kfs.coa.businessobject.SubObjectCode.class;
+            case KFSPropertyConstants.FINANCIAL_BALANCE_TYPE_CODE:
+                return org.kuali.kfs.coa.businessobject.BalanceType.class;
+            case KFSPropertyConstants.FINANCIAL_OBJECT_TYPE_CODE:
+                return org.kuali.kfs.coa.businessobject.ObjectType.class;
+            case KFSPropertyConstants.PROJECT_CODE:
+                return org.kuali.kfs.coa.businessobject.ProjectCode.class;
+            case KFSPropertyConstants.UNIVERSITY_FISCAL_PERIOD_CODE:
+                return org.kuali.kfs.coa.businessobject.AccountingPeriod.class;
+            case KFSPropertyConstants.FINANCIAL_DOCUMENT_TYPE_CODE:
+            case KFSPropertyConstants.REFERENCE_FINANCIAL_DOCUMENT_TYPE_CODE:
+                return org.kuali.rice.kew.doctype.bo.DocumentTypeEBO.class;
+        }
+        return null;
+    }
+
+    @SuppressWarnings("rawtypes")
+    private static String generateURLTitle(ResultRow row, Column column, Map<String, String> parameters) {
+        if (parameters.size() == 0) {
+            return KFSConstants.EMPTY_STRING;
+        }
+        Class clazz = getClassForField(column.getPropertyName());
+        if (clazz == null) {
+            return KFSConstants.EMPTY_STRING;
+        }
+        String className = clazz.getSimpleName();
+        if (className.equals("DocumentTypeEBO")) {

```

Comment:
These have been changed to used the value of the associated class's .getSimpleName() method.